src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -214,73 +221,86 @@ def _process_resource_change(
     retryer: Task | None = None,
     **kwargs: Any,
 ) -> None:
-    # The class is serialized as a string when enqueueing the class.
-    model: type[Event] | type[Model] = TYPES[sender]
-    instance: Event | Model | None = None
-
-    project_id: int | None = kwargs.get("project_id", None)
-    group_id: int | None = kwargs.get("group_id", None)
-    if sender == "Error" and project_id and group_id:
-        # Read event from nodestore as Events are heavy in task messages.
-        nodedata = nodestore.backend.get(Event.generate_node_id(project_id, str(instance_id)))
-        if not nodedata:
-            extra = {"sender": sender, "action": action, "event_id": instance_id}
-            logger.info("process_resource_change.event_missing_event", extra=extra)
-            return
-        instance = Event(
-            project_id=project_id, group_id=group_id, event_id=str(instance_id), data=nodedata
-        )
-        name = sender.lower()
-    else:
-        # Some resources are named differently than their model. eg. Group vs Issue.
-        # Looks up the human name for the model. Defaults to the model name.
-        name = RESOURCE_RENAMES.get(model.__name__, model.__name__.lower())
-
-    # By default, use Celery's `current_task` but allow a value to be passed for the
-    # bound Task.
-    retryer = retryer or current_task
-
-    # We may run into a race condition where this task executes before the
-    # transaction that creates the Group has committed.
-    if not issubclass(model, Event):
-        try:
-            instance = model.objects.get(id=instance_id)
-        except model.DoesNotExist as e:
-            # Explicitly requeue the task, so we don't report this to Sentry until
-            # we hit the max number of retries.
-            return retryer.retry(exc=e)
-
-    event = f"{name}.{action}"
-
-    if event not in VALID_EVENTS:
-        return
+    with SentryAppInteractionEvent(
+        operation_type=SentryAppInteractionType.PREPARE_WEBHOOK,
+        event_type=f"process_resource_change.{sender}_{action}",
+    ).capture() as lifecycle:
+
+        # The class is serialized as a string when enqueueing the class.
+        model: type[Event] | type[Model] = TYPES[sender]
+        instance: Event | Model | None = None
+
+        project_id: int | None = kwargs.get("project_id", None)
+        group_id: int | None = kwargs.get("group_id", None)
+        if sender == "Error" and project_id and group_id:
+            # Read event from nodestore as Events are heavy in task messages.
+            nodedata = nodestore.backend.get(Event.generate_node_id(project_id, str(instance_id)))
+            if not nodedata:
+                raise SentryAppSentryError(
+                    message=f"{SentryAppWebhookFailureReason.MISSING_EVENT}",
+                    webhook_context={"sender": sender, "action": action, "event_id": instance_id},
```

Comment:
is the `webhook_context` used if the logic is being executed in a task, not an endpoint? you should have `sender`, `action`, and `instance_id` in the sentry error since it's being passed into the `process_resource_change_bound` task

Reply:
`webhook_context` is being used akin to `extras` in logging. That's true we'll see those parts in the sentry error but it would be nice to have that in logging too

Reply:
removed the `webhook_context` since as you noted we already got that schtuff in the function params